From fd82454aeb6972926774d086ab4f8406361b4bef Mon Sep 17 00:00:00 2001
From: "yang.wang" <yang.wang@deepcomputing.io>
Date: Thu, 17 Apr 2025 11:31:33 +0800
Subject: [PATCH 285/540] update es8326 and i2s driver for audio format support

- Modify es8326_FORMATS to support S16_LE, S24_LE, and S32_LE,
  removing S20_3LE to resolve distortion issues.
- Fix i2s_hw_params to use CLOCK_CYCLES_24 for S24_LE/S24_3LE,
  correcting CCR configuration for 24-bit audio.

Signed-off-by: yang.wang <yang.wang@deepcomputing.io>
---
 sound/soc/codecs/es8326.c | 4 ++--
 sound/soc/eswin/esw-i2s.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/es8326.c b/sound/soc/codecs/es8326.c
index 55aa264d81653..1e252abaa6cb7 100644
--- a/sound/soc/codecs/es8326.c
+++ b/sound/soc/codecs/es8326.c
@@ -501,8 +501,8 @@ static int es8326_set_bias_level(struct snd_soc_component *codec,
 	return 0;
 }
 
-#define es8326_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE |\
-	SNDRV_PCM_FMTBIT_S24_LE)
+#define es8326_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S24_LE |\
+   SNDRV_PCM_FMTBIT_S32_LE)
 
 static const struct snd_soc_dai_ops es8326_ops = {
 	.hw_params = es8326_pcm_hw_params,
diff --git a/sound/soc/eswin/esw-i2s.c b/sound/soc/eswin/esw-i2s.c
index c71bc41f11f20..3c1a7a6aa08c0 100755
--- a/sound/soc/eswin/esw-i2s.c
+++ b/sound/soc/eswin/esw-i2s.c
@@ -461,7 +461,7 @@ static int i2s_hw_params(struct snd_pcm_substream *substream,
 	case SNDRV_PCM_FORMAT_S24_LE:
 	case SNDRV_PCM_FORMAT_S24_3LE:
 		config->data_width = 24;
-		i2s_drvdata->ccr = CLOCK_CYCLES_32 << CCR_WSS_POS |
+		i2s_drvdata->ccr = CLOCK_CYCLES_24 << CCR_WSS_POS |
 					NO_CLOCK_GATING;
 		i2s_drvdata->xfer_resolution = RESOLUTION_24_BIT;
 		break;
-- 
2.53.0

